cyipopt/scipy_interface.py

```diff
@@ -239,6 +251,12 @@ def intermediate(self, alg_mod, iter_count, obj_value, inf_pr, inf_du, mu,
                      ls_trials):
 
         self.nit = iter_count
+        if self.callback is not None:
+            iterate = self.get_current_iterate()
+            res = OptimizeResult()
+            res.x = iterate["x"]
+            res.fun = obj_value
+            self.callback(intermediate_result=res)
```

Comment:
@mdhaber To gracefully terminate, I believe all you have to do is return False here

Reply:
OK. I'll give that a shot.